api/hyper_block_proxy.go

```diff
@@ -2,31 +2,42 @@ package api
 
 import (
 	"encoding/hex"
+	"fmt"
 	"net/http"
 	"strconv"
 
 	"github.com/ElrondNetwork/covalent-indexer-go/cmd/proxy/config"
 	"github.com/gin-gonic/gin"
 )
 
+const (
+	startNonce = "startNonce"
+	endNonce   = "endNonce"
+)
+
 type hyperBlockProxy struct {
 	hyperBlockFacade HyperBlockFacadeHandler
 	options          config.HyperBlockQueryOptions
+	batchSize        uint32
 }
 
 // NewHyperBlockProxy will create a covalent proxy, able to fetch hyper block requests
 // from Elrond and return them in covalent format
 func NewHyperBlockProxy(
 	hyperBlockFacade HyperBlockFacadeHandler,
-	hyperBlockQueryOptions config.HyperBlockQueryOptions,
+	cfg *config.Config,
```

Comment:
no need pointer here

Reply:
Right, shouldn't have used pointer to config struct.
Removed it